### PR TITLE
feat(clients): add explicit unsubscribe API

### PR DESCRIPTION
**Pull request by niklasad1**
I reckon that sometimes one wants to explicitly wait until the unsubscribe method has been called which this does.

We might want to use tin the benches instead of the drop impl.

### CODE REVIEW

File: core/src/client/mod.rs [hunk 1]
```diff
@@ -203,12 +203,25 @@ impl<Notif> Subscription<Notif> {
...
 	}
 
 	/// Return the subscription type and, if applicable, ID.
 	pub fn kind(&self) -> &SubscriptionKind {
-		&self.kind
+		self.kind.as_ref().expect("only None after unsubscribe; qed")
+	}
+
+	/// Unsubscribe and consume the subscription.
+	pub async fn unsubscribe(mut self) -> Result<(), Error> {
```
Comment (jsdw) [inline core/src/client/mod.rs:215]: Is there a sensible way to test this?

**Review by jsdw — APPROVED** [review]
LGTM, and using `Option` there is def cleaner :)

File: core/src/client/mod.rs (hunk 1 above)
Comment (niklasad1) [reply core/src/client/mod.rs:215]: I think so, sorry but stuck on my other middleware PR :)

**Review by lexnv — APPROVED** [review]
LGTM! 👍

File: tests/tests/integration_tests.rs [hunk 2]
```diff
@@ -62,8 +70,40 @@ async fn ws_subscription_works() {
...
+
+	// It's technically possible to have race-conditions between the notifications and the unsubscribe message.
+	// So let's wait for the first notification and then unsubscribe.
+	let _item = sub.next().await.unwrap().unwrap();
+
+	sub.unsubscribe().await.unwrap();
+
+	let mut success = false;
+
+	// wait until the next slot is opened, as only one concurrent call is allowed.
```
Comment (niklasad1) [inline tests/tests/integration_tests.rs:91]:
```suggestion
	// Wait until the the slot available as only one concurrent call is allowed then we know that subscription call has
	// been finished
```

**Pull request merged**